integration/cdi/src/main/java/org/apache/cxf/cdi/JAXRSCdiResourceExtension.java

```diff
@@ -74,16 +74,16 @@
             featureBeans.add((Bean< ? extends Feature >)event.getBean());
         } else if (CdiBusBean.CXF.equals(event.getBean().getName())
                 && Bus.class.isAssignableFrom(event.getBean().getBeanClass())) {
-            busBean = event.getBean();
+            hasBus = true;
         }
     }
 
     public void load(@Observes final AfterDeploymentValidation event, final BeanManager beanManager) {
+        // no need of creational context, it only works for app scoped instances anyway
+        final Bean<?> busBean = beanManager.resolve(beanManager.getBeans(CdiBusBean.CXF));
         bus = (Bus)beanManager.getReference(
```

Comment:
@rmannibucau, could you please explain the difference between getting bean from the event (in `collect` method) and resolving the bean by name in the `load` method: `busBean = event.getBean()` vs `beanManager.resolve(beanManager.getBeans(CdiBusBean.CXF))`? I am not sure I understand the subtleties between those two.